PHPCompatibility/Sniffs/Interfaces/InternalInterfacesSniff.php

```diff
@@ -118,4 +156,37 @@ public function process(File $phpcsFile, $stackPtr)
             }
         }
     }
+
+
+    /**
+     * Processes this test for when a use token is encountered.
+     *
+     * - Save imported classes for later use.
+     *
+     * @since 10.0.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token in
+     *                                               the stack passed in $tokens.
+     *
+     * @return void
+     */
+    private function processUseToken(File $phpcsFile, $stackPtr)
+    {
+        if (!UseStatements::isImportUse($phpcsFile, $stackPtr)) {
+            return;
+        }
+
+        $splitUseStatement = UseStatements::splitImportUseStatement($phpcsFile, $stackPtr);
+
+        foreach ($splitUseStatement['name'] as $name => $fullyQualifiedName) {
+            $lowerFullyQualifiedName = strtolower($fullyQualifiedName);
+
+            if (isset($this->newClasses[$lowerFullyQualifiedName])) {
```

Comment:
```suggestion
            if (isset($this->internalInterfaces[$lowerFullyQualifiedName])) {
```

What is concerning about this is that the tests didn't fail on this. Is there anything which can be done about that ?

Reply:
This is indeed concerning. I will investigate.

Reply:
We simply had no test case were an internal PHP interface was imported via `use` and then used in a class declaration. Added the case and fixed the bug.